Design database

ArmineZakaryan opened the pull request.

ArmineZakaryan committed:
Design database

Tamara556 commented on src/main/java/org/example/coffeeshop/entity/Status.java:
```diff
@@ -0,0 +1,5 @@
+package org.example.coffeeshop.entity;
+
+public enum Status {
+    NEW, DELIVERED, CANCELED
```
CANCELLED

ArmineZakaryan committed:
Fix enum status

Tamara556 commented on pom.xml:
```diff
@@ -0,0 +1,133 @@
+        <relativePath/> <!-- lookup parent from repository -->
+    </parent>
+    <groupId>org.example</groupId>
+    <artifactId>Coffee-shop1</artifactId>
```
Coffee-shop, without 1

ArmineZakaryan committed:
Fix name Coffee-shop1 to Coffee-shop

ArmineZakaryan committed:
Change SQL language.
Replace all DTOs to Request/Response classes.
Replace int to Long in all entities.
Change enum type to varchar.

Astgh-Gevorgyan commented on src/main/java/org/example/coffeeshop/dto/CategoryDto.java:
```diff
@@ -0,0 +1,17 @@
+@Builder
+public class CategoryDto {
+
+    private int id;
```
Արմինե ջան ստեղ ուշադիր էղի, id-ն int ես տվել, բայց entity-ի մեջ Long

ArmineZakaryan replied:
Dto- ները հանելու եմ

Astgh-Gevorgyan commented on src/main/resources/db/migration/004_create_order_table.xml:
```diff
@@ -0,0 +1,25 @@
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+
+    <changeSet id="004_create_orders_table" author="azakaryan">
+        <createTable tableName="orders">
```
ստեղ orders ես գրել, բայց entity-ի մեջ order, մեկն ու մեկը փոխե

ArmineZakaryan committed:
Fix mistakes

Astgh-Gevorgyan commented on src/main/resources/db/changelog.xml:
```diff
@@ -0,0 +1,11 @@
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+
+    <include file="db/migration/001_create_user_table.xml"/>
```
ստեղ  օգտագործենք  include all, որ ամեն անգամ xml path  չտանք

Astgh-Gevorgyan commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+spring.application.name=Coffee-Shop
+spring.datasource.url=jdbc:mysql://localhost:3306/coffee_Shop?createDatabaseIfNotExist=true
```
ստեղ էլ shop

ArmineZakaryan replied:
PostgreSQL- ով անելուց դրանք փոխելու եմ

Astgh-Gevorgyan replied:
անունի համար եմ գրե, որ s-ն փոքրատառով

ArmineZakaryan replied:
Պրոյեկտի անունը մեծատառով է գրած, եդպես է սարքվել

Astgh-Gevorgyan replied:
jdbc:mysql://localhost:3306/coffee_**Shop**?createDatabaseIfNotExist=true 
ստեղած մասին է

ArmineZakaryan committed:
Change db
